lib/core/src/Cardano/Wallet/CoinSelection/Fee.hs

```diff
@@ -59,18 +60,52 @@ import Data.Word
 import GHC.Generics
     ( Generic )
 
-import qualified Codec.CBOR.Encoding as CBOR
-import qualified Codec.CBOR.Write as CBOR
-import qualified Data.ByteString as BS
-import qualified Data.ByteString.Lazy as BL
 import qualified Data.List as L
 
+
 {-------------------------------------------------------------------------------
-                                Fee Adjustment
+                                    Types
 -------------------------------------------------------------------------------}
 
 -- | A 'Fee', isomorph to 'Coin' but ease type-signatures and readability.
 newtype Fee = Fee { getFee :: Word64 }
+    deriving (Eq, Ord)
+
+-- | A linear equation on the transaction size. Represents the @\s -> a + b*s@
+-- function where @s@ is the transaction size in bytes, @a@ and @b@ are
+-- constant coefficients.
+data TxSizeLinear =
+    TxSizeLinear (Quantity "lovelace" Double) (Quantity "lovelace/byte" Double)
```

Comment:
why "lovelace" is in `Double`? Intuitively it should be integral type, not decimal.

Also, is `Double` the right type to pick here as it has arithmetic error?

Reply:
> why "lovelace" is in Double? Intuitively it should be integral type, not decimal.

aha - its a first coeficient in linear function that estimates end lovelace - but intermediate result will be rational. Did we pick `Double` because we will round at the end - so arithmetic error if it happens will be insignificant (non existent probably)?

Reply:
I am not sure why "we" picked double initially for the fees... It's quite common in financial system to have fees working at a fraction of the decimal but in the case of Ada which already has 6 decimals, I am not sure this was a wise decision. 
Anyway, we don't really have our say here because that's what currently run on the protocol.